shared/middleware/helpers/source_bucket.rb

```diff
@@ -66,6 +66,49 @@ def restore_previous_version(encrypted_channel_id, filename, version_id, user_id
     response.to_h
   end
 
+  # Copies the files in the src_channel to the dest_channel
+  # Update the animation manifest to include the version ids of the animations
+  # in dest_channel
+  # Note: this function assumes that the animations have already been copied
+  def remix_source(src_channel, dest_channel, animation_list)
+    src_owner_id, src_channel_id = storage_decrypt_channel_id(src_channel)
+    dest_owner_id, dest_channel_id = storage_decrypt_channel_id(dest_channel)
+
+    src_prefix = s3_path src_owner_id, src_channel_id
+
+    # For each file, copy it, update the animations, return it
+    s3.list_objects(bucket: @bucket, prefix: src_prefix).contents.map do |fileinfo|
+      filename = %r{#{src_prefix}(.+)$}.match(fileinfo.key)[1]
+
+      dest = s3_path dest_owner_id, dest_channel_id, filename
+
+      # Update animation manifest
+      key = s3_path src_owner_id, src_channel_id, filename
+      src_object = s3.get_object(bucket: @bucket, key: key)
+      src_body = src_object.body.read
+
+      if filename.casecmp?('main.json')
+        src_body = ProjectSourceJson.new(src_body)
+
+        if src_body.animation_manifest?
+          # Update the manifest to reference the newest version of the animations
+          # in the destination channel
+          src_body.each_animation do |a|
+            next if library_animation? a
+            anim_response = animation_list.select do |item|
```

Comment:
Since you only expect to find one matching entry in the animation list, [`Enumerable.find`](https://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-find) might be a better choice than `select` here.